system/win_main.cpp

```diff
@@ -59,7 +59,7 @@ int WINAPI WinMain (HINSTANCE hInstance, HINSTANCE /*hPrevInstance*/, LPSTR szCm
 #else
 int main(int argc, char** argv)
 {
-    return application(0).main(nullptr, 0);
+    return application(nullptr).main(*argv, 0);
```

Comment:
We'll probably want to just not compile this file at all on other platforms and create a new file specifically for OSX e.g. `osx_main.cpp`, since everything is stubbed out anyway. Can do this later though. Alternatively it might be easier to just use a framework like SDL on other platforms (or all platforms) so this would become `sdl_main.cpp`.

Reply:
I agree! I was just trying to bash the segfault at the time; this doesn't contribute in any other meaningful way.

I'll do whatever makes the most sense to you! Can discuss later.

Reply:
No changes required for now!